### ISSUE TITLE
Case sensitivity in model queries

### ISSUE THREAD

**Issue by treefort** [dchester/epilogue#77]
Hey! 

The documentation suggests that the standard `q` query functionality performs a case-insensitive substring search (ie: `resource?q=james` returns an object where `name:"James Conrad"`). In practice, I have not found this to be the case.

For instance, `resource?q=vin` returns records with `name:"Alvin"`, where `resource?q=Vin` yields an empty set.

I'm using the postgres dialect. Here are the relevant libraries/versions:
```
  "dependencies": {
    "epilogue": "^0.5.1",
    "express": "~4.12.4",
    "pg": "^4.4.0",
    "pg-hstore": "^2.3.2",
    "restify": "^3.0.3",
    "sequelize": "^3.3.0",
    "sequelize-cli": "^1.7.0"
  }
```

**Comment by mbroadst** [comment]
@treefort you're right, and I thought I could just change "like" to "$iLike" [here](https://github.com/dchester/epilogue/blob/master/lib/Controllers/list.js#L47), but it fails on a few tests. Unfortunately that's about all the time I have these day :smile:, any interest in trying to figure this one out? PRs are gladly welcomed

**Comment by treefort** [comment]
The operator could be passed in has an optional attribute in the search options. ex:

```javascript
var users = rest.resource({
    model: User,
    endpoints: ['/users', '/users/:id'],
    search: {
      param: 'searchOnlyUsernames',
      attributes: [ 'username' ],
      operator: '$iLike'    // (defaults to 'like'
    }
});
```
It's not pretty, but it'd work... plus it wouldn't mess up existing implementations. I can put it in a PR with corresponding tests and stuff, but I wanted to get your thoughts first.

**Comment by mbroadst** [comment]
@treefort I wouldn't be against it, but we should also conform with the docs right? (as in default to iLike). I know this seems sinfully lazy, but I can't even recall what tests broke - did you take a look?

**Comment by treefort** [comment]
Basically, any test that uses a substring match failed. If I'm understanding this correctly, sequelize isn't guarding against the fact that iLike is only supported in pg... so it seems like a strange default. Here's how I tested:

```javascript
User.findAll({where: { username: {$iLike: "%e%"}}})
```

which resulted in...

`Unhandled rejection SequelizeDatabaseError: SQLITE_ERROR: near "ILIKE": syntax error`

**Comment by mbroadst** [comment]
aah yeah of course, and mysql does case insensitive searches by default as well as sqlite. Yeah okay, please go ahead and add the operator for pg!

**Comment by treefort** [comment]
Awesome. Addressed with #78.
